common/ferrostar/src/models.rs

```diff
@@ -1,20 +1,33 @@
+use alloc::{string::String, vec::Vec};
 use geo::{Coord, LineString, Point, Rect};
+#[cfg(feature = "uniffi")]
 use polyline::encode_coordinates;
 use serde::Deserialize;
+
+#[cfg(feature = "std")]
 use std::time::SystemTime;
 
+#[cfg(feature = "wasm_js")]
+use web_time::SystemTime;
+
 #[cfg(test)]
 use serde::Serialize;
 use uuid::Uuid;
 
-#[derive(Debug, thiserror::Error, uniffi::Error)]
+#[derive(Debug)]
+#[cfg_attr(feature = "std", derive(thiserror::Error))]
+#[cfg_attr(feature = "uniffi", derive(uniffi::Error))]
 pub enum ModelError {
-    #[error("Failed to generate a polyline from route coordinates: {error}.")]
+    #[cfg_attr(
+        feature = "std",
+        error("Failed to generate a polyline from route coordinates: {error}.")
+    )]
```

Comment:
For `no_std` platforms, we cannot implement `Error` because the type is from `std`. However, in a few months (October), `Error` is [moving from `std` to `core`](https://github.com/orgs/time-rs/projects/1/views/1?pane=issue&itemId=15671948) :D